src/application.ts

```diff
@@ -102,25 +94,11 @@ export class Application {
    */
   public on (eventName: string | string[], callback: (context: Context) => void) {
     if (typeof eventName === 'string') {
-
       return this.events.on(eventName, async (event: Context) => {
-        const log = this.log.child({name: 'event', id: event.id})
-
         try {
-          let github
-
-          if (isUnauthenticatedEvent(event)) {
-            github = await this.auth()
-            log.debug('`context.github` is unauthenticated. See https://probot.github.io/docs/github-api/#unauthenticated-events')
-          } else {
-            github = await this.auth(event.payload.installation.id, log)
-          }
-
-          const context = new Context(event, github, log)
-
-          await callback(context)
+          await callback(await this.adapter.createContext(event))
```

Comment:
FIXME: move into `GitHubApp` when webhook is received so event emitter is just called with an instance of `Context`